src/js/index.js

```diff
@@ -396,6 +397,47 @@ function handleBoardDragEnd(e) {
   App Functions
 */
 
+function toggleTheme() {
+  if (theme === "light") {
+    document.body.className = theme = "dark";
+    setLocalStorageItem("theme", "dark");
+  } else {
+    document.body.className = "";
+    theme = "light";
+    setLocalStorageItem("theme", "light");
+  }
+
+  // Redraw the canvas
+  onResize();
+}
+
+function handleTheme() {
+  const savedPreference = getLocalStorageItem("theme");
+
+  // Prefer saved preference over OS preference
+  if (savedPreference) {
+    if (savedPreference === "dark") {
+      document.body.className = theme = "dark";
+      setLocalStorageItem("theme", "dark");
+    } else {
+      document.body.className = "";
+      theme = "light";
+      setLocalStorageItem("theme", "light");
+    }
+    return;
+  }
+
+  if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+    document.body.className = theme = "dark";
```

Comment:
So far I've been using classList.add/remove. Just worried if we don't use that method, things might break with this approach if <body> has additional classes appended to it in the future.

Reply:
Great! I actually prefer that too, should have checked if you were already using it :)